internal/usagegen/usagegen.go

```diff
@@ -255,3 +244,35 @@ func isDirectory(path string) bool {
 
 	return info.IsDir()
 }
+
+type fileSystem struct {
```

Comment:
Do we want this in speakeasy-core? It's used here and in openapi-generation.

Reply:
I kept them separate as the one in openapi-generation is for testing and I would expect these two filesystems will diverge over time